FirebaseAppDistribution/Sources/FIRFADApiService.m

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#import "FirebaseAppDistribution/Sources/FIRFADApiService.h"
+#import <Foundation/Foundation.h>
+#import "FirebaseAppDistribution/Sources/FIRFADLogger.h"
+#import "FirebaseCore/Sources/Private/FirebaseCoreInternal.h"
+#import "FirebaseInstallations/Source/Library/Private/FirebaseInstallationsInternal.h"
+
+NSString *const kFIRFADApiErrorDomain = @"com.firebase.appdistribution.api";
+NSString *const kFIRFADApiErrorDetailsKey = @"details";
+NSString *const kHTTPGet = @"GET";
+// The App Distribution Tester API endpoint used to retrieve releases
+NSString *const kReleasesEndpointURLTemplate =
+    @"https://firebaseapptesters.googleapis.com/v1alpha/devices/"
+    @"-/testerApps/%@/installations/%@/releases";
+NSString *const kInstallationAuthHeader = @"X-Goog-Firebase-Installations-Auth";
+NSString *const kApiHeaderKey = @"X-Goog-Api-Key";
+NSString *const kResponseReleasesKey = @"releases";
+
+@implementation FIRFADApiService
+
++ (void)generateAuthTokenWithCompletion:(FIRFADGenerateAuthTokenCompletion)completion {
+  FIRInstallations *installations = [FIRInstallations installations];
```

Comment:
The same here, [installationsWithApp:](https://github.com/firebase/firebase-ios-sdk/blob/0ed26aedb85f3f3fd37b0077133347183db5de7a/FirebaseInstallations/Source/Library/Public/FIRInstallations.h#L74) method should be used if App Distribution supports multiple apps.